zboxcore/sdk/deleteworker.go

```diff
@@ -81,41 +86,55 @@ func (req *DeleteRequest) getObjectTreeFromBlobber(blobber *blockchain.StorageNo
 }
 
 func (req *DeleteRequest) ProcessDelete() error {
-	numList := len(req.blobbers)
-	objectTreeRefs := make([]fileref.RefEntity, numList)
+	num := len(req.blobbers)
+	objectTreeRefs := make([]fileref.RefEntity, num)
+	removedNum := 0
 	req.wg = &sync.WaitGroup{}
-	req.wg.Add(numList)
-	for i := 0; i < numList; i++ {
+	req.wg.Add(num)
+	for i := 0; i < num; i++ {
 		go func(blobberIdx int) {
 			defer req.wg.Done()
 			refEntity, err := req.getObjectTreeFromBlobber(req.blobbers[blobberIdx])
-			if err != nil {
-				Logger.Error(err.Error())
+			if err == nil {
+				req.Lock()
+				req.consensus.consensus++
+				req.listMask |= (1 << uint32(blobberIdx))
+				objectTreeRefs[blobberIdx] = refEntity
```

Comment:
This objectTreeRefs[blobberIdx] does not have to be done in the req.Lock() I think.

Reply:
It is required. Because it is invoked in callback of multiple getobjecttree requests

Reply:
typo, it is invoked in callback of multiple goroutines in for